suite-native/atoms/src/Skeleton/BoxSkeleton.tsx

```diff
@@ -30,32 +36,45 @@ export const BoxSkeleton = ({
     const {
         utils: { colors },
     } = useNativeStyles();
+    const progress = useSharedValue(0);
 
-    // Offsetting the gradient center horizontally behind the rounded rect edges make the animations look smoother.
-    const gradientHorizontalOffset = width / GRADIENT_OFFSET_DIVISOR;
+    useEffect(() => {
+        progress.value = withRepeat(withTiming(width, { duration: ANIMATION_DURATION }), -1);
+    }, [width, progress]);
 
-    const gradientCenterHorizontalValue = useTiming(
-        { from: -gradientHorizontalOffset, to: width + gradientHorizontalOffset, loop: true },
-        { duration: ANIMATION_DURATION, easing: Easing.inOut(Easing.linear) },
-    );
+    const position = useDerivedValue(() => [
+        {
+            translateX: interpolate(progress.value, [0, width], [-width, width]),
+        },
+    ]);
+
+    const rct = useMemo(() => {
+        const coreRect = rect(0, 0, width, height);
+        return rrect(coreRect, borderRadius, borderRadius);
+    }, [width, height, borderRadius]);
 
-    const animatedGradientCenter = useComputedValue(
-        () => vec(gradientCenterHorizontalValue.current, height / 2),
-        [gradientCenterHorizontalValue],
+    const gradientColors = useMemo(
```

Comment:
is this memo helping? wouldnt LinearGradient rebuild anyway, because we insert new `vec` as start and end into it?

Reply:
it may rerender even if prop are not changed at all so its safer to have it there